src/eHealth.locations/services/selected-location-factory.js

```diff
@@ -246,8 +246,19 @@ angular.module('eHealth.locations.services')
             }
           });
           return angular.copy(adminDivisions);
+        },
+        clear: function () {
+          levels.forEach(function (level) { delete level.selected });
         }
       };
+
+      Object.defineProperty(location, 'innermost', {
```

Comment:
Well, it is a readonly property and we have `Object.defineProperty` now. But I can change that to a more traditional getter method, if you want.